File: 0566-reshape-the-matrix/0566-reshape-the-matrix.py
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        m = len(mat)
        n = len(mat[0])

        #예외처리
        if m * n != r * c:
            return mat
        
        #mat to list
        elements = []
        for i in mat:
            for element in i:
                elements.append(element)

        result = []
        idx = 0
        for _ in range(r):
            temp = []
            for _ in range(c):
                temp.append(elements[idx])
                idx += 1
            result.append(temp[:]) 

        return result

File: 0566-reshape-the-matrix/0566-reshape-the-matrix.java
class Solution {
    public int[][] matrixReshape(int[][] mat, int r, int c) {
        int m = mat.length;
        int n = mat[0].length;
        
        //exception
        if (m*n != r*c) return mat;
        
        int[][] result = new int[r][c];
        
        int out_r = 0;
        int out_c = 0;
        
        for (int i=0; i < m; i++) {
            
            for (int j=0; j < n; j++) {
                
                result[out_r][out_c] = mat[i][j];
                out_c++;
                
                if (out_c == c) {
                    out_r++;
                    out_c = 0;
                }
            }
        }
        
        return result;
        
    }
}